CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -0,0 +1,441 @@
+﻿using AutoMapper;
+using CharlieBackend.Business.Services;
+using CharlieBackend.Core.DTO.Schedule;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.Mapping;
+using CharlieBackend.Core.Models.ResultModel;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+using CharlieBackend.Business.Services.ScheduleServiceFolder;
+using FluentAssertions;
+using CharlieBackend.Business.Services.Interfaces;
+
+namespace CharlieBackend.Api.UnitTest
+{
+    public class ScheduleServiceTests : TestBase
+    {
+        private readonly IMapper _mapper;
+        private readonly IScheduledEventHandlerFactory _scheduledEventFactory;
+        private Mock<IScheduledEventRepository> _scheduleRepositoryMock;
+
+        public ScheduleServiceTests()
+        {
+            _mapper = GetMapper(new ModelMappingProfile());
+            _scheduledEventFactory = new ScheduledEventHandlerFactory();
+            _scheduleRepositoryMock = new Mock<IScheduledEventRepository>();
+            _unitOfWorkMock.Setup(x => x.ScheduledEventRepository).Returns(_scheduleRepositoryMock.Object);
+        }
+
+        private void Initialize(CreateScheduleDto createScheduleDto, Mock<IThemeRepository> themeRepositoryMock = null,
+            Mock<IMentorRepository> mentorRepositoryMock = null, Mock<IStudentGroupRepository> studentGroupRepositoryMock = null)
+        {
+            _scheduleRepositoryMock.Setup(x => x.AddRange(new List<ScheduledEvent>()
+            {
+                new ScheduledEvent
+                {
+                    ThemeId = 5,
+                    StudentGroupId = 3,
+                    MentorId = 1
+                }
+            }
+            ));
+
+            var eventOccuranceRepositoryMock = new Mock<IEventOccurrenceRepository>();
+
+            if (createScheduleDto != null)
+            {
+                eventOccuranceRepositoryMock.Setup(x => x.Add(
+                    new EventOccurrence
+                    {
+                        StudentGroupId = 3,
+                        EventStart = createScheduleDto.Range.StartDate,
+                        Pattern = createScheduleDto.Pattern.Type
+                    }
+                    ));
+            }
+
+            _unitOfWorkMock.Setup(x => x.EventOccurrenceRepository).Returns(eventOccuranceRepositoryMock.Object);
+
+            if (themeRepositoryMock != null)
+            {
+                _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(themeRepositoryMock.Object);
```

Comment:
Please create a read-only field for themeRepositoryMock, mentorRepositoryMock and other. Then set = new Mock<T>() to them in the constructor. Then setup 
 _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(_themeRepositoryMock.Object);
in the constuctor.
And then configure _themRepositoryMock in each method where you need.

Reply:
Ok, done